homeassistant/components/lock/__init__.py

```diff
@@ -129,8 +130,17 @@ def add_default_code(self, data: dict[Any, Any]) -> dict[Any, Any]:
         if not code:
             code = self._lock_option_default_code
         if self.code_format_cmp and not self.code_format_cmp.match(code):
-            raise ValueError(
-                f"Code '{code}' for locking {self.entity_id} doesn't match pattern {self.code_format}"
+            if TYPE_CHECKING:
+                assert self.code_format
+            raise ServiceValidationError(
+                f"Code '{code}' for locking {self.entity_id} doesn't match pattern {self.code_format}",
```

Comment:
Hmmm `code` is a password field, we should not show the contents of it I guess?

Reply:
Right, we shouldn't do that 👍